### PR TITLE
[phpstan] Add, enable first level and static fixes :tada: 

### PR DESCRIPTION
**Pull request by TomasVotruba**

### CODE REVIEW

**Comment by TomasVotruba** [comment]
Ready for review and merge :+1:

**Comment by TomasVotruba** [comment]
Poing

File: helpers/dataTypes.php [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
+ *
+ * @param mixed $obj object(s)
+ *
+ * @return array|mixed
  */
 function obj_to_array( $obj ) {
     // Not an array or object? Return back what was given
-    if( !is_array( $obj ) && !is_object( $obj ) ) 
+    if( !is_array( $obj ) && !is_object( $obj ) )
         return $obj;
```
Comment (TomasVotruba) [inline helpers/dataTypes.php:12]: The `$obj` type can be anything, and the return type also.

File: src/JSONDB.php [hunk 2]
```diff
@@ -106,13 +106,12 @@ private function check_file()
         // Check if its arrays of jSON
         if (! is_array($content) && is_object($content)) {
             throw new \Exception('An array of json is required: Json data enclosed with []');
-            return false;
```
Comment (TomasVotruba) [inline src/JSONDB.php:109]: The `throw` already finished the script, so this line is never run.

File: src/JSONDB.php [hunk 3]
```diff
@@ -106,13 +106,12 @@ private function check_file()
...
         }
         // An invalid jSON file
         elseif (! is_array($content) && ! is_object($content)) {
             throw new \Exception('json is invalid');
-            return false;
```
Comment (TomasVotruba) [inline src/JSONDB.php:114]: The `throw` already finished the script, so this line is never run.

File: src/JSONDB.php [hunk 4]
```diff
@@ -235,7 +234,7 @@ public function insert($file, array $values)
         // $this->last_indexes = [ ( count( $this->content ) - 1 ) ];
         $this->commit();
 
-        // return $this->last_indexes;
+        return $this->last_indexes;
```
Comment (TomasVotruba) [inline src/JSONDB.php:237]: To comply with return type defined in this method :+1:

File: src/JSONDB.php [hunk 5]
```diff
@@ -327,21 +326,21 @@ private function _update()
                     if (! array_diff_key($this->update, $content)) {
                         $this->content[$i] = (object) array_merge($content, $this->update);
                     } else {
-                        throw new Exception('Update method has an off key');
+                        throw new \Exception('Update method has an off key');
```
Comment (TomasVotruba) [inline src/JSONDB.php:329]: The previous `...\...\Exception` was missing. This is to use the native one :+1:

File: src/JSONDB.php [hunk 6]
```diff
@@ -381,8 +380,6 @@ public function trigger()
 
     /**
      * Flushes indexes they won't be reused on next action
-     *
-     * @return object $this
      */
     private function flush_indexes($flush_where = false)
```
Comment (TomasVotruba) [inline src/JSONDB.php:384]: This method has no return value

**Comment by TomasVotruba** [comment]
Ready for review & merge :heavy_check_mark:

File: src/JSONDB.php (hunk 4 above)
Comment (donjajo) [reply src/JSONDB.php:237]: Last indexes is no longer used. It was removed on purpose. On the changelog: https://github.com/donjajo/php-jsondb/releases/tag/v3.0

Comment (donjajo) [reply src/JSONDB.php:237]: I forgot to remove the comment. I will do now.

**Comment by TomasVotruba** [comment]
Hi, I'm checking this PR. What can I do to help to merge this one?
Thanks

**Comment by donjajo** [comment]
> Hi, I'm checking this PR. What can I do to help to merge this one? Thanks

Hello,
It has been a couple of busy weeks, and from a glance, aside from the return value I mentioned above, they look good.
I will look thoroughly this weekend hopefully and merge.
I just want to make sure there are no breaking changes within this sub-version

**Comment by donjajo** [comment]
This has been merged: https://github.com/donjajo/php-jsondb/commit/0890b34878e93f5b1cdbfdc505265b4b1eb467aa

**Comment by TomasVotruba** [comment]
Thanks for finishg it :slightly_smiling_face:

**Pull request closed without merging**